src/legacy/core_plugins/kibana/public/dashboard/grid/dashboard_grid.tsx

```diff
@@ -161,54 +194,56 @@ class DashboardGridUi extends React.Component {
     });
   }
 
-  createEmbeddableFactoriesMap(panels) {
+  public createEmbeddableFactoriesMap(panels: PanelMap) {
     Object.values(panels).map(panel => {
       if (!this.embeddableFactoryMap[panel.type]) {
         this.embeddableFactoryMap[panel.type] = this.props.getEmbeddableFactory(panel.type);
       }
     });
   }
 
-  componentWillMount() {
+  public componentWillMount() {
     this.createEmbeddableFactoriesMap(this.props.panels);
   }
 
-  componentWillReceiveProps(nextProps) {
+  public componentWillReceiveProps(nextProps: Props) {
     this.createEmbeddableFactoriesMap(nextProps.panels);
   }
 
-  onLayoutChange = (layout) => {
+  public onLayoutChange = (layout: Layout[]) => {
     const { onPanelsUpdated, panels } = this.props;
-    const updatedPanels = layout.reduce((updatedPanelsAcc, panelLayout) => {
-      updatedPanelsAcc[panelLayout.i] = {
-        ...panels[panelLayout.i],
-        panelIndex: panelLayout.i,
-        gridData: _.pick(panelLayout, ['x', 'y', 'w', 'h', 'i'])
-      };
-      return updatedPanelsAcc;
-    }, []);
+    const updatedPanels = layout.reduce(
+      (updatedPanelsAcc, panelLayout) => {
+        updatedPanelsAcc[panelLayout.i] = {
+          ...panels[panelLayout.i],
+          panelIndex: panelLayout.i,
+          gridData: _.pick(panelLayout, ['x', 'y', 'w', 'h', 'i']),
+          embeddableConfig: {},
```

Comment:
@stacey-gammon This seems to be the difficult part in the PR and I thought I should run it past your eyes.

The short version is that `onLayoutChange` takes `Layout[]` and returns `PanelStateMap`. Either A) we need to convert between the two or B) there is something wrong with one or both of the types. 

I'm fairly certain that `Layout[]` is correct since it relates directly to `React-Grid-Layout` functionality. I'm fairly certain that it needs to return a `PanelStateMap` (formerly PanelMap, but I changed the name since it had PanelStates and not Panels) since that eventually gets passed into the action. I'm now supplying embeddableConfig and id which feels like a bad idea....but it passes type_check so it must be good, right?